### DISCUSSION TITLE
q-tree: click customized header (with a button) without triggering node itself

### DISCUSSION THREAD

**Discussion by 549393092** [quasarframework/quasar#11048] [General - Components / Directives / etc] [answered]
Dear,

I created a tree menu with a lot of menus. I would like to add a button in the each tree node, so when user click the button, it will add it as a favorate.
So I customize the q-tree header by below example.
https://github.com/quasarframework/quasar/blob/dev/docs/src/examples/QTree/SlotsDefault.vue 
```
<q-tree
      :nodes="customize"
      node-key="label"
      default-expand-all
    >
      <template v-slot:default-header="prop">
        <div class="row items-center">
          <q-icon :name="prop.node.icon || 'share'" color="orange" size="28px" class="q-mr-sm" />
          <div class="text-weight-bold text-primary">{{ prop.node.label }}</div>
          <q-btn icon=“star” @click="addToFavorate(prop.node)"/>
        </div>
      </template>
    </q-tree>
```
When I click the button, the event will be triggered. However, the tree node itself will be triggered as well. How can we avoid it? Many thanks!

**Comment by rstoenescu** [comment, marked as answer]
You just need to stop the propagation of the event.

`<q-btn @click.stop="....`

**Reply by 549393092** [reply]
@rstoenescu Thank you! It works.
